core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala

```diff
@@ -387,6 +402,162 @@ class MultiCloudWorkspaceServiceSpec extends AnyFlatSpec with Matchers with Test
   }
 
+  behavior of "cloneMultiCloudWorkspace"
+
+  it should "fail to clone an azure workspace until [WOR-625]" in {
+    val workspaceManagerDAO = new MockWorkspaceManagerDAO()
+    val config = MultiCloudWorkspaceConfig(ConfigFactory.load())
+    val bpmDAO = mock[BillingProfileManagerDAO](RETURNS_SMART_NULLS)
+
+    val billingProject = RawlsBillingProject(
+      RawlsBillingProjectName("fake_mc_billing_project_name"),
+      CreationStatuses.Ready,
+      None,
+      None,
+      billingProfileId = Some(UUID.randomUUID().toString)
+    )
+
+    val mcWorkspaceService = spy(
+      MultiCloudWorkspaceService.constructor(
+        slickDataSource,
+        workspaceManagerDAO,
+        bpmDAO,
+        new MockSamDAO(slickDataSource),
+        config,
+        workbenchMetricBaseName
+      )(testContext)
+    )
+
+    doReturn(Future.successful(billingProject))
+      .when(mcWorkspaceService)
+      .getBillingProjectContext(any(), any())
+
+    doReturn(Future.successful(testData.azureWorkspace))
+      .when(mcWorkspaceService)
+      .getWorkspaceContext(equalTo(testData.azureWorkspace.toWorkspaceName), any())
+
+    val result = intercept[RawlsExceptionWithErrorReport] {
+      Await.result(
+        mcWorkspaceService.cloneMultiCloudWorkspace(
+          mock[WorkspaceService](RETURNS_SMART_NULLS),
+          testData.azureWorkspace.toWorkspaceName,
+          WorkspaceRequest(
+            "fake_mc_billing_project_name",
+            UUID.randomUUID().toString,
+            Map.empty,
+            None,
+            None,
+            None,
+            None
+          )
+        ),
+        Duration.Inf
+      )
+    }
+
+    result.errorReport.statusCode.value shouldBe StatusCodes.NotImplemented
+  }
+
+  it should "not allow users to clone azure workspaces into gcp billing projects" in {
+    val workspaceManagerDAO = new MockWorkspaceManagerDAO()
+    val config = MultiCloudWorkspaceConfig(ConfigFactory.load())
+    val bpmDAO = mock[BillingProfileManagerDAO](RETURNS_SMART_NULLS)
+
+    val mcWorkspaceService = spy(
+      MultiCloudWorkspaceService.constructor(
+        slickDataSource,
+        workspaceManagerDAO,
+        bpmDAO,
+        new MockSamDAO(slickDataSource),
+        config,
+        workbenchMetricBaseName
+      )(testContext)
+    )
+
+    doReturn(Future.successful(testData.billingProject))
+      .when(mcWorkspaceService)
+      .getBillingProjectContext(any(), any())
+
+    doReturn(Future.successful(testData.azureWorkspace))
+      .when(mcWorkspaceService)
+      .getWorkspaceContext(equalTo(testData.azureWorkspace.toWorkspaceName), any())
+
+    val result = intercept[RawlsExceptionWithErrorReport] {
+      Await.result(
+        mcWorkspaceService.cloneMultiCloudWorkspace(
+          mock[WorkspaceService](RETURNS_SMART_NULLS),
+          testData.azureWorkspace.toWorkspaceName,
+          WorkspaceRequest(
+            testData.billingProject.projectName.value,
```

Comment:
Is this billingProject a GCP billing project? I'm having trouble when reading the test knowing that we are trying to clone to a GCP workspace.

Reply:
Actually, the same billing project is used below (https://github.com/broadinstitute/rawls/pull/2142/files#diff-b8196c4af84188faa6eb1f65595077eb0b25f6c06f5a8e64b5c60e4832b93118R545), so I'm even more confused.